fuel-client/tests/coin.rs

```diff
@@ -89,3 +91,115 @@ async fn first_5_coins() {
+                owner,
+                amount: i as Word,
+                color: if i <= 5 {
+                    Color::new([1u8; 32])
+                } else {
+                    Default::default()
+                },
+                maturity: Default::default(),
+                status: CoinStatus::Unspent,
+                block_created: Default::default(),
+            };
+
+            let utxo_id = UtxoId {
+                tx_id: Bytes32::from([i as u8; 32]),
+                output_index: 0,
+            };
+            (utxo_id.into(), coin)
+        })
+        .collect();
+
+    let mut db = Database::default();
+    for (id, coin) in coins {
+        Storage::<Bytes32, Coin>::insert(&mut db, &id, &coin).unwrap();
+    }
+
+    // setup server & client
+    let srv = FuelService::from_database(db, Config::local_node())
+        .await
+        .unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    // run test
+    let coins = client
+        .coins(
+            format!("{:#x}", owner).as_str(),
+            Some(format!("{:#x}", Color::new([1u8; 32])).as_str()),
+            PaginationRequest {
+                cursor: None,
+                results: 10,
+                direction: PageDirection::Forward,
+            },
+        )
+        .await
+        .unwrap();
+    assert!(!coins.results.is_empty());
+    assert_eq!(coins.results.len(), 5)
+}
+
+#[tokio::test]
+async fn only_spent_coins() {
+    let owner = Address::default();
+
+    // setup test data in the node
+    let coins: Vec<(Bytes32, Coin)> = (1..10usize)
+        .map(|i| {
+            let coin = Coin {
+                owner,
+                amount: i as Word,
+                color: Default::default(),
+                maturity: Default::default(),
+                status: if i <= 5 {
+                    CoinStatus::Unspent
+                } else {
+                    CoinStatus::Spent
+                },
+                block_created: Default::default(),
+            };
+
+            let utxo_id = UtxoId {
+                tx_id: Bytes32::from([i as u8; 32]),
+                output_index: 0,
+            };
+            (utxo_id.into(), coin)
+        })
+        .collect();
+
+    let mut db = Database::default();
+    for (id, coin) in coins {
+        Storage::<Bytes32, Coin>::insert(&mut db, &id, &coin).unwrap();
+    }
+
+    // setup server & client
+    let srv = FuelService::from_database(db, Config::local_node())
+        .await
+        .unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    // run test
+    let coins = client
+        .coins(
+            format!("{:#x}", owner).as_str(),
+            None,
+            PaginationRequest {
+                cursor: None,
+                results: 10,
+                direction: PageDirection::Forward,
+            },
+        )
+        .await
+        .unwrap();
+    assert!(!coins.results.is_empty());
+    assert_eq!(coins.results.len(), 5)
```

Comment:
It's not clear to me if the test is expecting these to be unspent or only spent coins

Reply:
The naming was wrong on that one, it's expecting unspent. I fixed that also added new assertions: https://github.com/FuelLabs/fuel-core/pull/100/commits/0dc03d1c727b99a7a648e4ea03db05f2f4050a97